### PR TITLE
#12451 migrate components/delete_modal_trigger

### PR DESCRIPTION
**Pull request by QamarFarooq**
#### Ticket Link : https://mattermost.atlassian.net/browse/MM-18977
Fixes https://github.com/mattermost/mattermost-server/issues/12451

### CODE REVIEW

**Comment by hanzei** [comment]
Hi @QamarFarooq,

There are some linter errors with your code:
```bash

/root/mattermost-webapp/components/delete_modal_trigger.test.tsx
  151:1  error  Trailing spaces not allowed  no-trailing-spaces

/root/mattermost-webapp/components/delete_modal_trigger.tsx
   1:1   error  Too many blank lines at the beginning of file. Max of 1 allowed  no-multiple-empty-lines
   6:1   error  There should be at least one empty line between import groups    import/order
   8:1   error  There should be at least one empty line between import groups    import/order
  18:1   error  More than 1 blank line not allowed                               no-multiple-empty-lines
  21:82  error  Multiple spaces found before '{'                                 no-multi-spaces
  21:84  error  Block must not be padded by blank lines                          padded-blocks
  21:85  error  Trailing spaces not allowed                                      no-trailing-spaces
  26:30  error  Expected '===' and instead saw '=='                              eqeqeq
  26:54  error  Trailing spaces not allowed                                      no-trailing-spaces
  30:45  error  A space is required after ','                                    comma-spacing
  37:45  error  A space is required after ','                                    comma-spacing
  38:6   error  Trailing spaces not allowed                                      no-trailing-spaces
  45:46  error  A space is required after ','                                    comma-spacing

```

Let me know, if you need help resolving them

**Comment by jasonblais** [comment]
Hi @QamarFarooq, are you open to help with the linter errors? If you have any questions, don't hesitate to let us know :)

**Comment by hanzei** [comment]
@QamarFarooq The CI reports:
```
/root/mattermost-webapp/components/delete_modal_trigger.test.tsx
   14:48  error  Missing space before =>      arrow-spacing
   14:51  error  Missing space after =>       arrow-spacing
   17:48  error  Missing space before =>      arrow-spacing
   17:51  error  Missing space after =>       arrow-spacing
  155:1   error  Trailing spaces not allowed  no-trailing-spaces

/root/mattermost-webapp/components/delete_modal_trigger.tsx
   4:1   error  There should be at least one empty line between import groups  import/order
   6:1   error  There should be at least one empty line between import groups  import/order
  17:82  error  Multiple spaces found before '{'                               no-multi-spaces
  17:84  error  Block must not be padded by blank lines                        padded-blocks
  17:85  error  Trailing spaces not allowed                                    no-trailing-spaces
  21:30  error  Expected '===' and instead saw '=='                            eqeqeq
  21:54  error  Trailing spaces not allowed                                    no-trailing-spaces
  25:45  error  A space is required after ','                                  comma-spacing
  32:45  error  A space is required after ','                                  comma-spacing
```

Let me know if you need help resolving them

**Comment by hanzei** [comment]
@QamarFarooq is this ready for review?

**Comment by QamarFarooq** [comment]
> @QamarFarooq is this ready for review?

Yes it is

File: components/delete_modal_trigger.tsx [hunk 1]
```diff
@@ -1,54 +1,66 @@
...
 
-    handleKeyDown = (e) => {
+    handleKeyDown = (e: React.KeyboardEvent) => {
         if (Utils.isKeyPressed(e, Constants.KeyCodes.ENTER)) {
-            this.handleConfirm(e);
+            this.handleConfirm();
         }
     }
 
+    get triggerTitle(): JSX.Element {
```
Comment (catalintomai) [inline components/delete_modal_trigger.tsx]: Where is it element (and the ones below) being used?

Comment (devinbinnie) [inline components/delete_modal_trigger.tsx]: I don't think these are required, these look like optional props that weren't specified in the original `PropTypes` definition. Can we just add these as optional props and remove these definitions?

File: components/delete_modal_trigger.test.tsx [hunk 2]
```diff
@@ -5,16 +5,16 @@ import React from 'react';
...
 import Constants from 'utils/constants';
-import DeleteModalTrigger from 'components/delete_modal_trigger.jsx';
+import DeleteModalTrigger from 'components/delete_modal_trigger';
 import ConfirmModal from 'components/confirm_modal.jsx';
 
 describe('components/DeleteModalTrigger', () => {
     test('should throw error when trying to construct DeleteModalTrigger', () => {
         expect(() => {
-            new DeleteModalTrigger(); //eslint-disable-line no-new
+            new DeleteModalTrigger({onDelete: () => {}}); //eslint-disable-line no-new
```
Comment (devinbinnie) [inline components/delete_modal_trigger.test.tsx]:
```suggestion
            new DeleteModalTrigger({onDelete: jest.fn()}); //eslint-disable-line no-new
```

Can we do this for consistency?

Comment (QamarFarooq) [reply components/delete_modal_trigger.test.tsx]: Sorry, i had to reinstall windows cuz i had a virus, and i was trying to set up windows server setup, but it staking forever, just going to go with Windows WSL instead. I will get to it asap

File: components/delete_modal_trigger.tsx (hunk 1 above)
Comment (QamarFarooq) [reply components/delete_modal_trigger.tsx]: These are class props, if i remove them than line 67 compalins about `Property 'triggerTitle' does not exist on type 'DeleteModalTrigger'.ts(2339)`

**Comment by QamarFarooq** [comment]
@devinbinnie This is ready for review

File: components/delete_modal_trigger.test.tsx [hunk 3]
```diff
@@ -71,50 +71,22 @@ describe('components/DeleteModalTrigger', () => {
             />
         );
 
-        wrapper.find(ConfirmModal).first().props().onConfirm();
+        wrapper.find(ConfirmModal).first().props().onConfirm(true);
         expect(onDelete).toHaveBeenCalledTimes(1);
     });
 
-    test('should have called onDelete on enter key down', () => {
```
Comment (devinbinnie) [inline components/delete_modal_trigger.test.tsx]: How come these tests were removed?

Comment (QamarFarooq) [reply components/delete_modal_trigger.test.tsx]: @devinbinnie looks like `onKeyDown` was passed to confirm_modal.tsx but was not being used inside confirm_modal.tsx. Which is why i removed these tests as that handler was never being invoked.

https://github.com/mattermost/mattermost-webapp/blob/master/components/confirm_modal.tsx#L8

**Comment by QamarFarooq** [comment]
@devinbinnie Can you remove the 'stale' and 'awaiting submitter' labels?

**Comment by jasonblais** [comment]
@DHaussermann & @catalintomai reminder to help with dev/QA reviews, thanks!

**Comment by DHaussermann** [comment]
@devinbinnie or @catalintomai  Can you please advise on which delete modals in the app used `delete_modal_trigger` to provide some guidance for testing?

**Comment by DHaussermann** [comment]
@devinbinnie or @catalintomai can you clarify for me what  area of the application is affected by this change?

**Comment by devinbinnie** [comment]
@DHaussermann It's used by the Custom Emoji page. If you create a Custom Emoji and then in the list click 'Delete', this file represents the modal that pops up confirming if you want to delete it.

**Comment by devinbinnie** [comment]
@QamarFarooq can you merge your branch with `master` and fix the conflicts so we can merge? Thanks :)

**Comment by devinbinnie** [comment]
@QamarFarooq Are you able to help out with fixing the merge conflicts here? Let us know :)

**Comment by jasonblais** [comment]
@devinbinnie wondering if this is something we can do to unblock the contributor?

**Pull request merged**